index.bs

```diff
@@ -199,23 +199,35 @@ Abstract Operations {#abstract-operations}
 
 Automation {#automation}
 ==========
-This section extends the [=automation=] section defined in the Generic Sensor API [[GENERIC-SENSOR]]
-to provide mocking information about the proximity level for the purposes of
-testing a user agent's implementation of {{ProximitySensor}} API.
 
-<h3 id="mock-proximity-sensor-type">Mock Sensor Type</h3>
+This section extends [[GENERIC-SENSOR#automation]] by providing [=Proximity Sensor=]-specific virtual sensor metadata.
 
-The {{ProximitySensor}} class has an associated [=mock sensor type=] which is
-<a for="MockSensorType" enum-value>"proximity"</a>, its [=mock sensor reading values=]
-dictionary is defined as follows:
+The [=per-type virtual sensor metadata=] [=map=] must have the following [=map/entry=]:
+: [=map/key=]
+:: "`proximity`"
+: [=map/value=]
+:: A [=virtual sensor metadata=] whose [=virtual sensor metadata/virtual sensor type=] is [=Proximity Sensor=] and [=reading parsing algorithm=] is the [=proximity reading parsing algorithm=].
 
-<pre class="idl">
-  dictionary ProximityReadingValues {
-    required double? distance;
-    required double? max;
-    required boolean? near;
-  };
-</pre>
+<h3 dfn>Proximity reading parsing algorithm</h3>
+<div algorithm="proximity reading parsing algorithm">
+    : input
+    :: |parameters|, a JSON {{Object}}
+    : output
+    :: A [=sensor reading=] or **undefined**
+
+    1.  Let |reading| be a new [=sensor reading=].
+    1.  Let |keys| be the [=/list=] « "`distance`", "`max`" ».
+    1.  [=map/For each=] |keys| of [=/list=]
```

Comment:
* "list" in https://infra.spec.whatwg.org/#list-iterate was the name of the list, not the type.
* You are still referring to map's "for each", not list's.
* Think of each like a "for each" that iterates through a list or a vector: you need the vector and the item that will receive each item in the vector.

Reply:
I think

```
- "list" in https://infra.spec.whatwg.org/#list-iterate was the name of the list, not the type.
- You are still referring to map's "for each", not list's.
```

are fixed in https://github.com/w3c/proximity/pull/57/commits/8eb1313cf653d65668c523d4b9e9831356f61e73
But doubt that 

```
Think of each like a "for each" that iterates through a list or a vector: you need the vector and the item that will receive each item in the vector.
```

is... Let's see